kafka/producer.go

```diff
@@ -197,8 +203,10 @@ func (p *Producer) Close() error {
 // the messages have been stored in the producer's buffer.
 func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 	ctx, span := p.tracer.Start(ctx, "producer.ProcessBatch", trace.WithAttributes(
+		attribute.Bool("sync", p.cfg.Sync),
 		attribute.Int("batch.size", len(*batch)),
 	))
+	defer span.End()
```

Comment:
So, I know I've brought this back and reverted #102.
But there were 2 issues with that PR:

We can't end the span within `Produce`, because it runs once for each event, which means the span is closed after the first event instead of the last.

If this method is async, it effectively returns quick, even though things are still running but in the background. (child spans can still be created, they will just show up as running after this method finished). Waiting for every child span to finish instead of the method to return gives a wrong visualization of what's happening.

Reply:
Should we also add the k/v Paris from `queuecontext.MetadataFromContext`? There may be useful contextual information in those.

Reply:
We definitely can. I'm always a bit wary of adding attributes without knowing what they are.
That metadata never includes any PII or sensitive data?